mule-extensions-api/src/main/java/org/mule/runtime/extension/api/util/ExtensionMetadataTypeUtils.java

```diff
@@ -344,20 +348,12 @@ public static boolean isReferableType(MetadataType type) {
    *
    * @since 1.4.0
    */
-  public static boolean areTypesEqual(MetadataType type, MetadataType otherType, ClassLoader classLoader) {
-    String typeClassName = getClassInformationName(type);
-    String otherTypeClassName = getClassInformationName(otherType);
+  public static boolean areTypesEqual(MetadataType type, MetadataType otherType) {
+    ClassInformationAnnotation typeClassInformation = type.getAnnotation(ClassInformationAnnotation.class).orElse(null);
+    ClassInformationAnnotation otherTypeClassInformation = otherType.getAnnotation(ClassInformationAnnotation.class).orElse(null);
 
-    if (typeClassName != null && otherTypeClassName != null) {
-      if (classLoader != null) {
-        if (!areAssignable(typeClassName, otherTypeClassName, classLoader)) {
-          return false;
-        }
-      } else {
-        if (!Objects.equals(typeClassName, otherTypeClassName)) {
-          return false;
-        }
-      }
+    if (typeClassInformation != null) {
+      return typeClassInformation.equals(otherTypeClassInformation);
```

Comment:
I think we should rely in the Id, if the class informations are different, we can say that the types are different. But if they are the same, I think we should keep checking the Id.